integration/container/stop_test.go

```diff
@@ -18,6 +18,47 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func TestStopContainerWithRestartPolicyAlways(t *testing.T) {
+	defer setupTest(t)()
+	client := request.NewAPIClient(t)
+	ctx := context.Background()
+
+	names := []string{"verifyRestart1", "verifyRestart2"}
+	for _, name := range names {
+		foo, err := client.ContainerCreate(ctx,
+			&container.Config{
+				Cmd:   []string{"false"},
```

Comment:
I don't really understand the behaviour of the old test.

It's running a container with command `false`, so the container is going to exit almost immediately anyway. Polling on `running` means it's going to attempt to inspect before the command is done running.

Then it runs `stop` and doesn't check any state.

What is this supposed to be testing? By the test name I would assume it is supposed to test that `docker stop` actually stops a container from restarting , even with `restart: always`.

So shouldn't this test:
* poll that the container is running, failed, or started. Any of those states should be fine, right? Anything but `created` ?
* after the `stop` check that the container is actually in the "done" state and isn't being restarted anymore